src/LightInject.Interception.Tests/ProxyBuilderTests.cs

```diff
@@ -246,6 +246,34 @@ public void Execute_NonGenericMethod_PassesMethodInfoToInterceptor()
 
             interceptorMock.Verify(i => i.Invoke(It.Is<IInvocationInfo>(ii => ii.Method == expectedMethod)));
         }
+        
+        [Fact]
+        public void Execute_NonGenericMethod_PassesMethodInvocationTargetInfoToInterceptorWhenProxyIsInterface()
+        {
+            var interceptorMock = new Mock<IInterceptor>();
+            var instance = CreateProxy<IClassWithOneMethod, ClassWithOneMethod>(interceptorMock.Object, "FirstMethod");
+            MethodInfo expectedMethod = typeof(IClassWithOneMethod).GetMethod("FirstMethod");
+            MethodInfo expectedMethodInvocationTarget = typeof(ClassWithOneMethod).GetMethod("FirstMethod");
+            
+            instance.FirstMethod();
+
+            interceptorMock.Verify(
+                i => i.Invoke(
+                    It.Is<IInvocationInfo>(ii => ii.Method == expectedMethod && ii.MethodInvocationTarget == expectedMethodInvocationTarget)));
+        }
+        
+        [Fact]
+        public void Execute_NonGenericMethod_PassesMethodInvocationTargetInfoToInterceptorWhenProxyIsNotInterface()
+        {
+            var interceptorMock = new Mock<IInterceptor>();
+            var instance = CreateProxy<IMethodWithNoParameters>(interceptorMock.Object);
+            MethodInfo expectedMethod = typeof(IMethodWithNoParameters).GetMethod("Execute");
+
+            instance.Execute();
+
+            interceptorMock.Verify(
+                i => i.Invoke(It.Is<IInvocationInfo>(ii => ii.Method == expectedMethod && ii.MethodInvocationTarget == expectedMethod)));
+        }    
```

Comment:
We could use a test for a generic method as well 👍

Reply:
Sure! I missed that. I added the tests, and thanks to you for this great library!